examples/functions/sentiment-analysis/index.ts

```diff
@@ -0,0 +1,46 @@
+import {createClient} from '@sanity/client'
+import {documentEventHandler} from '@sanity/functions'
+
+export const handler = documentEventHandler(async ({context, event}) => {
+  const client = createClient({
+    ...context.clientOptions,
+    apiVersion: 'vX', // vX currently required for Agent Actions
+    useCdn: false,
+  })
+  const {data} = event
+  const {local} = context // local is true when running locally
+
+  try {
+    const result = await client.agent.action.generate({
+      noWrite: Boolean(local), // if local is true, we don't want to write to the document, just return the result for logging
+
+      // Define which field to analyze
+      instructionParams: {
+        review: {
+          type: 'field',
+          path: 'review',
+        },
+      },
+
+      instruction: `Analyze the sentiment of the $review and categorize it into one of these 5 levels: very_positive, positive, neutral, negative, very_negative. Consider the emotional tone, word choice, and overall sentiment expressed in the text. Return only the sentiment level as a string.`,
```

Comment:
maybe non-critical, but I think it would be interesting to also draft some like actionable response as well? vs just a sentiment, is there another step that makes this part of a real workflow of like `0 out of 10 triggers x` that or we add that to the readme for like `how to handle sentiment levels` with some suggestive pathways. 

another possible flow would be to `review the review` read our internal FAQs and try to draft a response, could be like `our FAQs list a 8 hour battery life which is beyond the industry standard, if you're experience significantly less it could be faulty,..` me scope creeping.